feat(cli): add `mastra studio` command

NikAiyer opened the pull request:
## Description

<!-- Provide a brief description of the changes in this PR -->
Adds a new `mastra studio` CLI command that serves the built playground as a static server. This allows users to view the Mastra Studio UI after running `mastra build`, without needing to run the full dev server.


### Usage
```bash
mastra studio                    # Starts on default port 3000
mastra studio -p 8080            # Custom port
mastra studio -e .env.local      # Custom env file
```

- Added studio command to CLI with --port and --env options
- Validates that .mastra/output/playground exists before starting
- Uses locally installed serve binary instead of npx
- Properly handles process exit for both normal and error cases

## Related Issue(s)

<!-- Link to the issue(s) this PR addresses, using hashtag notation: #123 -->

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test update

## Checklist

- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a new "studio" CLI command to serve the built playground as a static site (configurable port, .env support).
  * New playground configuration guard and form with a Mastra instance URL input and improved config handling.
  * New instance-status check hook and broader routing/layout updates, including additional tools, workflows, and agents pages.

* **Chores**
  * Added runtime deps for the static server and dev type declarations; included changesets for a patch release.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

NikAiyer committed:
studio command for cli

NikAiyer committed:
Merge branch 'main' into feat/studio-command

NikAiyer committed:
update studio command

NikAiyer committed:
update path

NikAiyer committed:
add changeset

NikAiyer committed:
Merge branch 'main' into feat/studio-command

NikAiyer committed:
lint

NikAiyer committed:
merge

NikAiyer committed:
Merge branch 'main' into feat/studio-command

mfrachet commented:
I was having this one opened for a while: https://github.com/mastra-ai/mastra/pull/9938 . Do you want me to take over from this PR and add the UIs?

mfrachet committed:
Merge branches 'feat/studio-command' and 'main' of github.com:mastra-ai/mastra into feat/studio-command

mfrachet committed:
Merge branch 'main' into feat/studio-command

mfrachet committed:
move to serve-handler

mfrachet committed:
Merge branch 'feat/studio-command' of github.com:mastra-ai/mastra into feat/studio-command

NikAiyer committed:
Merge branch 'main' into feat/studio-command

mfrachet committed:
feat: studio command UI (#10966)

## Description

This pull request introduces a configuration guard flow for the
Playground app, ensuring users are prompted to configure a Mastra
instance URL before accessing the main application. It adds a new hook
to check the status of the Mastra instance, updates the configuration
context to use this status, and reorganizes the app's provider structure
for improved reliability and user experience.

**Configuration flow improvements:**

* Added a new `PlaygroundConfigGuard` component that prompts users to
enter the Mastra instance URL if no valid configuration is detected.
[[1]](diffhunk://#diff-f9dc89ace32467e76518d6bba51084578fefa28643a5f2dc3dd312019d75eb9eR1-R15)
[[2]](diffhunk://#diff-3171a51285be921f3b565cb578c50b385cf05d8ab80f737840d0426d77348065R4)
[[3]](diffhunk://#diff-c717bf55bad3b8f1b23b5cf8710cb699e37cef9f0c29697cacf8fda1b814ea06R82-L83)
* Updated the `StudioConfigForm` to include an input for the instance
URL and to save it as part of the configuration.
[[1]](diffhunk://#diff-e3ee069b5a79a83ddc1f8c0e054fc54f5e7654dbbdfc21576f091e83e58ca889R9)
[[2]](diffhunk://#diff-e3ee069b5a79a83ddc1f8c0e054fc54f5e7654dbbdfc21576f091e83e58ca889R27)
[[3]](diffhunk://#diff-e3ee069b5a79a83ddc1f8c0e054fc54f5e7654dbbdfc21576f091e83e58ca889L34-R36)
[[4]](diffhunk://#diff-e3ee069b5a79a83ddc1f8c0e054fc54f5e7654dbbdfc21576f091e83e58ca889R50-R57)

**Mastra instance status detection:**

* Introduced the `useMastraInstanceStatus` hook, which checks if the
provided Mastra instance endpoint is active using a network request.
* Enhanced the `StudioConfigProvider` to leverage the instance status,
automatically updating configuration state and preventing access to the
app if the instance is inactive.
[[1]](diffhunk://#diff-ff76ff42b00c858a03ba7cac143bf600f316b145ab432dd93a68f4d493ccc28aR3)
[[2]](diffhunk://#diff-ff76ff42b00c858a03ba7cac143bf600f316b145ab432dd93a68f4d493ccc28aR23-R41)
[[3]](diffhunk://#diff-ff76ff42b00c858a03ba7cac143bf600f316b145ab432dd93a68f4d493ccc28aR51-R56)

**App provider structure changes:**

* Moved the `PlaygroundQueryClient` provider to wrap the entire app,
ensuring query context is available throughout, and adjusted the
provider hierarchy for correct configuration flow.

## Related Issue(s)

<!-- Link to the issue(s) this PR addresses, using hashtag notation:
#123 -->

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test update

## Checklist

- [ ] I have made corresponding changes to the documentation (if
applicable)
- [ ] I have added tests that prove my fix is effective or that my
feature works

NikAiyer committed:
Merge branch 'main' into feat/studio-command

mfrachet reviewed (approved):
🤝
